0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1201 @@
+# Common Ledger API
+
+The Common Ledger API is a RESTful API served by a ledger (or an adapter), which provides functionality necessary for ILP compatibility. The Common Ledger API provides a single standard API that a ledger can serve in order to ease integration with other Interledger Protocol components and applications, such as the reference ILP Client and ILP Connector. This is not the only way a ledger can become ILP-enabled, but it provides a template that minimizes the integration work necessary for compatibility with ILP software.
+
+## Overview
+
+The Common Ledger API defines the following data types and structures:
+
+- [Transfer resource][]
+- [Account resource][]
```

Comment:
I agree that blockchains would need a proxy server to host a Common Ledger API, which would sit between the client application (wallet) and the blockchain server / p2p network. I'm not sure there's a way around that.

The "account information" returned can be quite minimal. In a blockchain case, "Get Account" could use the pubkey as the `name` and then just provide the known balance that can be spent by that key pair.

Maybe we need a special value for balances that are unknown in order to support this. I'm not really sure.